widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java

```diff
@@ -747,19 +747,29 @@ private void appendMessageItem(List<ChatItem> currentChatItems, ChatMessage mess
             appendSentMessage(currentChatItems, message);
             addVisitorAttachmentItemsToChatItems(currentChatItems, message);
         } else if (message.getSender() == Chat.Participant.OPERATOR) {
-            changeLastOperatorMessages(currentChatItems, message);
-            appendMessagesNotSeen();
-            if (message.getAttachment() instanceof SingleChoiceAttachment) {
-                SingleChoiceAttachment attachment = ((SingleChoiceAttachment) message.getAttachment());
-                emitViewState(
-                        chatState.chatInputModeChanged(
-                                attachment.getOptions() != null ?
-                                        ChatInputMode.SINGLE_CHOICE_CARD :
-                                        ChatInputMode.ENABLED
-                        )
-                );
-            }
+            onOperatorMessageReceived(currentChatItems, message);
+        }
+    }
+
+    private void onOperatorMessageReceived(List<ChatItem> currentChatItems, ChatMessage message) {
+        changeLastOperatorMessages(currentChatItems, message);
+        appendMessagesNotSeen();
+        changeChatInputMode(message);
+    }
+
+    private void changeChatInputMode(ChatMessage message) {
+        ChatInputMode newInputMode = getChatInputMode(message);
+        if (chatState.chatInputMode != newInputMode)
```

Comment:
should use curly braces in multiline if-else blocks. official standard.

Reply:
https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements

Reply:
fixed (also for chatinputmode method)

Reply:
beautiful